PROTO_tests/tests/indexing_test.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+NUM_TYPES = [ak.int64, ak.uint64, ak.float64, ak.bool, ak.bigint]
+BIGINT_AND_UINT_TYPES = [ak.bigint, ak.uint64]
+
+
+def key_arrays(size):
+    ikeys = ak.arange(size)
+    ukeys = ak.arange(size, dtype=ak.uint64)
+    return ikeys, ukeys
+
+
+def value_array(num_types, size):
+    array_dict = {
+        ak.int64: (i := ak.randint(0, size, size)),
+        ak.uint64: (u := ak.cast(i, ak.uint64)),
+        ak.float64: ak.array(np.random.randn(size)),
+        ak.bool: (i % 2) == 0,
+        ak.bigint: ak.cast(u, ak.bigint),
+        ak.str_: ak.cast(i, str),
+    }
+    return array_dict[num_types]
```

Comment:
this is actually a pretty inefficient way of doing this if we are only ever returning one array at a time. Because for every dtype we are calculating all these arrays even if we only use one

Also we should switch the var name to be `dtype` (prob the function name too) and pick more representative examples for each dtype
```python
def make_array(dtype, size):
    if dtype is ak.int64:
        return ak.randint(-size, size, size)
    elif dtype is ak.uint64:
        return ak.randint(0, size, size, dtype=dtype)
    elif dtype is ak.float64:
        return ak.randint(-size, size, size, dtype=dtype)
    elif dtype is ak.bool:
        return (ak.randint(0, size, size) % 2) == 0
    elif dtype is ak.bigint:
        return ak.randint(0, size, size, dtype=ak.uint64) + 2**200
    elif dtype is ak.str_:
        return ak.random_strings_uniform(1, 16, size=size)
    return None
```

Reply:
Updated